src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/PatchFunction.cs

```diff
@@ -218,7 +218,9 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
 
                 foreach (var typedName in curType.GetNames(DPath.Root))
                 {
-                    if (!tableType.TryGetFieldType(typedName.Name, out var displayName, out _))
+                    var xx = tableType.TryGetFieldType(typedName.Name, out var displayName, out _);
```

Comment:
Why is TryGetFieldType insufficient here?

Reply:
It will (potentially) scan the complete dataverse type tree while we don't even need the type 

This is a huge perf improvement

Reply:
ah, gotcha. This is a consequence of Dataverse relationships not being implemented via a derived record type. The laziness would be more directly accomplished by doing that, but I understand if that's out of scope right now. Glad the new method is internal, so we can move in that direction later.